packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1199,3 +1311,45 @@ class FollowerLayer extends ContainerLayer {
     properties.add(new TransformProperty('transform', getLastTransform(), defaultValue: null));
   }
 }
+
+/// A composited layer which annotates its children with a value.
+///
+/// These values can be retrieved using [Layer.find] with a given [Offset]. If
+/// a [Size] is provided to this layer, then find will check if the provided
+/// offset is within the bounds of the layer.
+class AnnotatedRegionLayer<T> extends ContainerLayer {
+  /// Creates a new annotated layer.
```

Comment:
These docs don't match our style guide.

Reply:
(Because they don't tell you anything you can't tell from the identifiers. Maybe mention the contract on size, e.g. that if null it won't clip, or something.)

Reply:
maybe say value can't be null? and assert it

Reply:
Fixed